Hazel/src/Hazel/Application.h

```diff
@@ -39,9 +40,10 @@ namespace Hazel {
 		float m_LastFrameTime = 0.0f;
 	private:
 		static Application* s_Instance;
+		std::string m_base_directory;
```

Comment:
Why not use `std::filesystem::path`? 

Then all your appends (`p1 / p2`) can be simplified for path can be simplified, tbh with the 3 of them I'd probably unroll the loop and check them explicitly commenting the reason for each.

Reply:
The parameter is turned into `std::filesystem::path` within `std::filesystem::exists()`. I don't see a prototype for `char*` so maybe C++17 automatically uses a `path` operator to do a cast-conversion? I don't see the advantage for rewriting this segment. I'm new to C++17 which is why I needed to use the new `for()` iterator.